### ISSUE TITLE
unawaited_futures lint does not work on the following line of code

### ISSUE THREAD

**Issue by nshahan** [dart-lang/linter#1007]
```dart
final cssOutput = sass.compileStringAsync(
        await buildStep.readAsString(inputId),
        importers: [new BuildImporter(buildStep)],
        style: _getValidOutputStyle());
```

You can see the signature of `compileStringAsync()` at the link below. It is async and not being awaited but the lint does not give any feedback about it.

https://github.com/sass/dart-sass/blob/8e1d80c4790a9a10afa3ecd0791f8611d5b53b6b/lib/sass.dart#L207

**Comment by natebosch** [comment]
Here's the full context:
 https://github.com/dart-league/sass_builder/blob/ac821bca17c207c0333c032e563eae2bfd0b537f/lib/sass_builder.dart#L41-L49

This is hitting the case where assigning to a variable suppresses the lint. _Usually_ that's an OK thing to suppress on because either that variable is unused or the later usage should statically warn you that you've got a `Future<T>` instead of a `T`... The problem comes in that this variable was used, just not in a way that is sane for a `Future<T>` but without any warning.

Maybe we should consider a lint like `avoid_to_string_on_future` which would have helped track this down?

**Comment by srawlins** [comment]
I think that's a fair assessment. Looks like the rule is working as best as we intend it to.
